src/framework/control.ml

```diff
@@ -422,7 +422,13 @@ struct
       let save_run = let o = get_string "save_run" in if o = "" then (if gobview then "run" else "") else o in
 
       let lh, gh = if load_run <> "" then (
-          let module S2' = (GlobSolverFromEqSolver (Generic.LoadRunIncrSolver (PostSolverArg))) (EQSys) (LHT) (GHT) in
+          let module PostSolverArg2 =
+          struct
+            include PostSolverArg
+            let should_prune = false (* the run should have been already pruned *)
```

Comment:
Should have indeed, but pruning a pruned solution shouldn't delete everything either, so this might be indicative of a deeper issue. If a solution is loaded, then the postsolver should be doing complete reverification anyway to check its validity and emit the warnings (by corresponding postsolvers), so it should still be reaching all the nodes and not pruning them.

Reply:
Just a thought, maybe the postsolver runs correctly on the loaded solution to check it and prints its warnings. While doing that it also collects a `reachable` set, which should prevent everything from being pruned. But maybe due to some hashconsing relifting problem, the `reachable` set lookups all fail, despite the values being in there or something.

Reply:
The issue is caused by the fact that in that `reachable` set, the node's id are not the same as in the one that are unmarshalled, which cause the pruning to remove everything.

Reply:
That's unusual because the postsolver should add to the reachable set the same nodes that it handles when iterating the loaded solution.
Maybe something related to this Gobview change? https://github.com/goblint/gobview/commit/f8c509dd3938d209cefe99b4e4dabbcdec173103

Reply:
Okay so commenting out the code of goblint/gobview@f8c509d does indeed restore the correct node id, but it does not solve the problem that the pruning remove everything.
So the way gobview load a solution must differ from the way of goblint

Reply:
Oh damn! I automatically thought you were already testing with that change.
Turns out that commit was on the `master` branch in Gobview's repository, but the gobview submodule in this Goblint's repository hadn't yet been updated to include that commenting out. I now did that though.

I suppose one thing to try would be disabling `ana.opt.hashcons` to see if it might have anything to do with hashconsing and relifting.

Reply:
Disabling `ana.opt.hashcons` does indeed solve the issue, but I don't think the issue comes directly from it, as one can execute a query with `ana.opt.hashcons` enabled and the `load_run` option set to some directory containing the results from `goblint`, and it works as expected.

My best guess is that there is something different between `goblint`'s initialization and `gobview`'s one, which I cannot find.

Reply:
Does GobView already perform `relift` on the loaded data, so that the hashcons tables are populated again?
Similar to what Goblint already does for the `load_run` option.
https://github.com/goblint/analyzer/blob/fe5c98d05915a39974cd50aa611eb6f66292eeb5/src/framework/control.ml#L440-L442